docs/utils/threading/worker-pool.md

```diff
@@ -148,3 +148,217 @@ functionality in Thunder, which will be further described in a different documen
+            }
+```
+The Scheduler uses the `Timer` class of the `Workerpool` class as a privileged class. How the `Timer` class is built and how it works can be seen in the `Timer.h` file.
+
+### Job class scheduling methods
+Each `job` has two methods by which we can reschedule them for future execution and cancel them.
+ These methods are:
+
+* `Reschedule()`
+* `Revoke()`
+
+#### Rescheduling job
+`Reschedule()` allows the scheduled `job` execution time to be changed. The id argument takes the time by which we should postpone the execution of the job. The method returns `true` if the operation is successful.
+```cpp
+bool Reschedule(const Core::Time& time)
+            {
+                bool rescheduled = false;
+
+                Core::ProxyType<IDispatch> job(ThreadPool::JobType<IMPLEMENTATION>::Reschedule(time));
+
+                if (job.IsValid() == true) {
+
+                    job = ThreadPool::JobType<IMPLEMENTATION>::Revoke();
+
+                    if (job.IsValid() == true) {
+                        rescheduled = (IWorkerPool::Instance().Revoke(job) == Core::ERROR_NONE);
+                        ThreadPool::JobType<IMPLEMENTATION>::Revoked();
+                    }
+
+                    job = (ThreadPool::JobType<IMPLEMENTATION>::Idle());
+
+                    if (job.IsValid() == true) {
+                        IWorkerPool::Instance().Schedule(time, job);
+                    }
+                }
+
+                return (rescheduled && (ThreadPool::JobType<IMPLEMENTATION>::IsIdle() == false));
+            }
+```
+!!!warning
+    The `Job` must first be submitted to be rescheduled!
+
+#### Revoking job
+`Revoke()` is used to revoke the execution of `job`.
+```cpp
+void Revoke()
+            {
+                Core::ProxyType<IDispatch> job(ThreadPool::JobType<IMPLEMENTATION>::Revoke());
+
+                if (job.IsValid() == true) {
+                    Core::IWorkerPool::Instance().Revoke(job);
+                    ThreadPool::JobType<IMPLEMENTATION>::Revoked();
+                }
+            }
+```
+
+#### Example
+Below is an example of the use of the `Reschedule()` method in the `ChannelMap` class in the `PluginServer.h` file.
+```cpp
+uint32_t Open(const uint32_t waitTime, const uint16_t connectionCheckTimer) {
+                _connectionCheckTimer = connectionCheckTimer * 1000;
+                if (connectionCheckTimer != 0) {
+                    _job.Reschedule(Core::Time::Now().Add(_connectionCheckTimer).Ticks());
+                }
+                return(BaseClass::Open(waitTime));
+            }
+```
+
+`Reschedule` method is used also in `ThunderNanoServices` repository. Here is few examples:
+
+`TimeSync/NTPClient.cpp`
+```cpp
+// See if we need rescheduling
+        if (result != Core::infinite) {
+            Core::Time timestamp(Core::Time::Now());
+            timestamp.Add(result);
+            _job.Reschedule(timestamp);
+```
+
+`ProcessMonitor/ProcessMonitor.h`
+```cpp
+void ScheduleJob()
```

Comment:
1. It is really nice that you include example from actual use cases, much better than including tests which only call the methods! But once again - if you include any code examples, please write a few sentences of what is going on in this code.
2. The main purpose to this documentation was to create a section for Scheduler. Maybe I am mistaken, but I don't think you included any use examples from Scheduler, only examples of methods from the templated class `JobType`, which is already talked about in the Workerpool documentation: https://github.com/rdkcentral/Thunder/blob/master/docs/utils/threading/worker-pool.md?plain=1#L51